Modules/_zstd/_zstdmodule.c

```diff
@@ -149,20 +146,15 @@ set_parameter_error(const _zstd_state* const state, int is_compress,
         bounds = ZSTD_dParam_getBounds(key_v);
     }
     if (ZSTD_isError(bounds.error)) {
-        PyErr_Format(state->ZstdError,
-                     "Invalid zstd %s parameter \"%s\".",
+        PyErr_Format(PyExc_ValueError, "invalid %s parameter '%s'",
                      type, name);
         return;
     }
 
     /* Error message */
-    PyErr_Format(state->ZstdError,
-                 "Error when setting zstd %s parameter \"%s\", it "
-                 "should %d <= value <= %d, provided value is %d. "
-                 "(%d-bit build)",
-                 type, name,
-                 bounds.lowerBound, bounds.upperBound, value_v,
-                 8*(int)sizeof(Py_ssize_t));
+    PyErr_Format(PyExc_ValueError,
+        "%d not in valid range %d <= value <= %d for %s parameter '%s'",
+        value_v, bounds.lowerBound, bounds.upperBound, type, name);
```

Comment:
I somewhat liked the original (ignoring grammatical mistakes). Perhaps a combination would be acceptable?
```suggestion
    PyErr_Format(PyExc_ValueError,
        "Error when setting %s parameter \"%s\", the provided value %d is not in "
        "the valid range %d <= value <= %d.",
        type, name, value_v, bounds.lowerBound, bounds.upperBound);
```

Reply:
"error when" is semi-redundant, this is an exception message. As an alternate?:

```suggestion
    PyErr_Format(PyExc_ValueError,
        "%s parameter '%s' recieved an illegal value %d; "
        "the valid range is %d <= value <= %d",
        type, name, value_v, bounds.lowerBound, bounds.upperBound);
```

Reply:
That works for me!